yt_dlp/extractor/allstar.py

```diff
@@ -0,0 +1,259 @@
+import functools
+import json
+
+from .common import InfoExtractor
+from ..utils import (
+    ExtractorError,
+    OnDemandPagedList,
+    UnsupportedError,
+    int_or_none,
+    parse_qs,
+    str_or_none,
+    traverse_obj,
+    urljoin,
+)
+
+
+_FIELDS = '''
+    _id
+    clipImageThumb
+    clipLink
+    clipLinkHLS
+    clipTitle
+    createdDate
+    shareId
+    user { _id }
+    username
+    views'''
+
+_EXTRA_FIELDS = '''
+    clipLength
+    clipSizeBytes
+    game'''
+
+_QUERIES = {
+    'clip': '''query ($id: String!) {
+        video: getClip(clipIdentifier: $id) {
+            %s %s
+        }
+    }''' % (_FIELDS, _EXTRA_FIELDS),
+    'montage': '''query ($id: String!) {
+        video: getMontage(clipIdentifier: $id) {
+            %s
+        }
+    }''' % _FIELDS,
+    'Clips': '''query ($page: Int!, $user: String!, $game: Int) {
+        videos: clips(search: createdDate, page: $page, user: $user, mobile: false, game: $game) {
+            data { %s %s }
+        }
+    }''' % (_FIELDS, _EXTRA_FIELDS),
+    'Montages': '''query ($page: Int!, $user: String!) {
+        videos: montages(search: createdDate, page: $page, user: $user) {
+            data { %s }
+        }
+    }''' % _FIELDS,
+    'Mobile Clips': '''query ($page: Int!, $user: String!) {
+        videos: clips(search: createdDate, page: $page, user: $user, mobile: true) {
+            data { %s %s }
+        }
+    }''' % (_FIELDS, _EXTRA_FIELDS),
+}
+
+
+class AllstarBaseIE(InfoExtractor):
+    @staticmethod
+    def _parse_video_data(video_data):
+        def _media_url_or_none(path):
+            return urljoin('https://media.allstar.gg/', str_or_none(path))
+
+        def _profile_url_or_none(path):
+            return urljoin('https:/allstar.gg/u/', str_or_none(path))
+
+        return traverse_obj(video_data, {
+            'id': ('_id', {str_or_none}),
+            'display_id': ('shareId', {str_or_none}),
+            'title': ('clipTitle', {str_or_none}),
+            'url': ('clipLink', {_media_url_or_none}),
+            'thumbnail': ('clipImageThumb', {_media_url_or_none}),
+            'duration': ('clipLength', {int_or_none}),
+            'filesize': ('clipSizeBytes', {int_or_none}),
+            'timestamp': ('createdDate', {int_or_none}),
+            'uploader': ('username', {str_or_none}),
+            'uploader_id': ('user', '_id', {str_or_none}),
+            'uploader_url': ('user', '_id', {_profile_url_or_none}),
+            'view_count': ('views', {int_or_none}),
```

Comment:
If you just want to ensure that a value is of a certain type, you can use `str` and `int`, instead of `str_or_none`/`int_or_none`.
```suggestion
            'id': ('_id', {str}),
            'display_id': ('shareId', {str}),
            'title': ('clipTitle', {str}),
            'url': ('clipLink', {_media_url_or_none}),
            'thumbnail': ('clipImageThumb', {_media_url_or_none}),
            'duration': ('clipLength', {int}),
            'filesize': ('clipSizeBytes', {int}),
            'timestamp': ('createdDate', {int_or_none}),
            'uploader': ('username', {str}),
            'uploader_id': ('user', '_id', {str}),
            'uploader_url': ('user', '_id', {_profile_url_or_none}),
            'view_count': ('views', {int}),
```

Reply:
`int_or_none` is appropriate for `duration` and `filesize`, if the value is given by the site as string and we need to convert to `int`

`str_or_none` may also be appropriate to use for `id` for the opposite reason. But I agree with the rest of the `str` changes

Reply:
`duration` and `filesize` are given as ints and `id` is given as a string

Reply:
In that case, then you are right that `str_or_none` shouldn't be used at all

`int_or_none` can't hurt though, it's fine to leave as-is

Reply:
> If you just want to ensure that a value is of a certain type, you can use `str` and `int`, instead of `str_or_none`/`int_or_none`.

thanks for the info.